src/test.rs

```diff
@@ -558,3 +559,68 @@ fn accept_one_byte_at_a_time() {
 
     t.join().unwrap();
 }
+
+#[test]
+fn split_cert_key() {
+    let listener = TcpListener::bind("127.0.0.1:0").unwrap();
+    let addr = listener.local_addr().unwrap();
+    let t = thread::spawn(move || {
+        let cert = include_bytes!("../test/cert.der");
+        let mut store = Memory::new().unwrap();
+        store.add_encoded_certificate(cert).unwrap();
+        let store = store.into_store();
+
+        let stream = TcpStream::connect(&addr).unwrap();
+        let creds = SchannelCred::builder()
+                                 .acquire(Direction::Outbound).unwrap();
+        let mut stream = tls_stream::Builder::new()
+            .domain("foobar.com")
+            .cert_store(store)
+            .connect(creds, stream)
+            .unwrap();
+        stream.write_all(&[1, 2, 3, 4]).unwrap();
+        stream.flush().unwrap();
+        assert_eq!(stream.read(&mut [0; 1024]).unwrap(), 4);
+        stream.shutdown().unwrap();
+    });
+
+    let cert = include_bytes!("../test/cert.der");
+    let cert = CertContext::new(cert).unwrap();
+
+    let mut options = AcquireOptions::new();
+    options.container("schannel-test");
```

Comment:
[MSDN/CryptAcquireContext](https://msdn.microsoft.com/en-us/library/windows/desktop/aa379886(v=vs.85).aspx):
>> Instead, either prevent key storage by passing the CRYPT_VERIFYCONTEXT flag in the dwFlags parameter, or use an application-specific container that is unlikely to be used by another application. [...] For performance reasons, we recommend that you set the pszContainer parameter to NULL and the dwFlags parameter to CRYPT_VERIFYCONTEXT in all situations where you do not require a persisted key.

That sounds to me like we want `verify_context()` here.